### ISSUE TITLE
In a directory with broken symlinks, not all files are listed

### ISSUE THREAD

**Issue by raj-magesh** [nvim-telescope/telescope-file-browser.nvim#129]
### Description

When I try to list files in a directory with broken symlinks, only those found before the first broken symlink are listed. For example, if the files are F_1, F_2, ..., F_k, ... F_n where F_k is a broken symlink, only the files F_1, F_2, ..., F_{k-1} are listed in Telescope.

### Neovim version

```markdown
NVIM v0.7.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser
```

### Operating system and version

Arch Linux x86_64. kernel 5.17.5-arch1-1

### Steps to reproduce

1. Create a folder `test` with files `f1` `f2` `f4` and a symbolic link `f3` to elsewhere.  
```
touch f3
mkdir test
ln -s f3 test/f3
touch test/f1 test/f2 test/f4
```

2. Delete the symlink target and open Neovim in the folder `test`
```
rm f3
cd test
nvim
```

3. Run `:Telescope file_browser`

### Expected behavior

All files `f1` through `f4` should be listed.

### Actual behavior

Only files `f1` and `f2` are listed, because `f3` is a broken link.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          'nvim-telescope/telescope-file-browser.nvim',
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('file_browser')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by miversen33** [comment]
It appears this issue has compounded with a recent PR.
Broken symlinks now cause an error in Telescope which prevents _anything_ from loading.
> [telescope] [WARN  08:13:08] ...ck/packer/start/telescope.nvim/lua/telescope/pickers.lua:489: Finder failed with msg:  ...im/lua/telescope/_extensions/file_browser/make_entry.lua:18: attempt
 to index field 'stat' (a nil value)
 
[The code snippet in question](https://github.com/nvim-telescope/telescope-file-browser.nvim/blob/ac22b741cd80bc6dfbd412955a9388a03e7a9f09/lua/telescope/_extensions/file_browser/make_entry.lua#L16)
```lua
local SIZE = {
  width = 7,
  right_justify = true,
  display = function(entry)
    local size = entry.stat.size -- stat doesn't exist on broken file symlinks it seems
``` 

Note: This will likely _also_ happen in [the DATE fetcher](https://github.com/nvim-telescope/telescope-file-browser.nvim/blob/ac22b741cd80bc6dfbd412955a9388a03e7a9f09/lua/telescope/_extensions/file_browser/make_entry.lua#L34) as it utilizes the same style of logic
```lua
local DATE = {
  width = 13,
  right_justify = true,
  display = function(entry)
    local mtime = entry.stat.mtime.sec
```

I noticed these errors as well while playing around with stat overriding. Most likely what is happening is plenary is not providing any details when stat is attached to entry (I can't remember where this happens off the top of my head but I can peek a bit when I get home).
I am unsure what about any previous PRs broke this as I dont believe this area of the code was touched in either #138 or #137 . 

Its probably not safe to assume stat exists here (or maybe better to force an empty table in place of stat if a failure is received from Plenary?)

I might take a crack at a PR on this tonight as its been a pain in my own project too

**Comment by miversen33** [comment]
Its no problem! The plugin is amazing and in my case (as we discussed on whatever issue number I have been exploring stat stuff), the issues I have been seeing are self inflicted. But that does mean that its probably worth getting it addressed at a higher level :) And I like this project alot so I might as well contribute instead of just complaining ;)

**Comment by fdschmidt93** [comment]
Should be fixed via #139 

Please let me otherwise know if any related issues persist.
